internal/overlord/logstate/syslog.go

```diff
@@ -0,0 +1,248 @@
+package logstate
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/canonical/pebble/internal/servicelog"
+)
+
+const maxLogBytes = 100 * 1024
+
+// SyslogWriter takes writes and formats them according to RFC5424.  The formatted syslog messages
+// are then forwarded to the specified underlying destination io.Writer.
+type SyslogWriter struct {
+	mu      sync.RWMutex
```

Comment:
Yeah - SetParam is intended to be called before writes start - I'd be fine documenting that method as not concurrency safe.  But the pid isn't known until after we start the command - so it is possible that early writes could have a wrong pid.  I'm not low-level savvy enough to know how to avoid that problem - so I added the mutex to at least limit the bad outcomes to at most a write with stale pid.